Common/Data/Consolidators/VolumeRenkoConsolidator.cs

```diff
@@ -0,0 +1,289 @@
+                var high = ((TradeBar)data).High;
+                var low = ((TradeBar)data).Low;
+                var close = ((TradeBar)data).Close;
+
+                if (_firstTick)
+                {
+                    _firstTick = false;
+
+                    OpenOn = data.Time;
+                    CloseOn = data.EndTime;
+                    OpenRate = open;
+                    HighRate = high;
+                    LowRate = low;
+                    CloseRate = close;
+                }
+                else
+                {
+                    CloseOn = data.EndTime;
+
+                    if (high > HighRate)
+                    {
+                        HighRate = high;
+                    }
+
+                    if (low < LowRate)
+                    {
+                        LowRate = low;
+                    }
+
+                    CloseRate = close;
+                }
+            }
+
+            _volumeLeftOver = Next(data, volume);
+        }
+
+        /// <summary>
+        /// Scans this consolidator to see if it should emit a bar due to time passing
+        /// </summary>
+        /// <param name="currentLocalTime">The current time in the local time zone (same as <see cref="BaseData.Time"/>)</param>
+        public void Scan(DateTime currentLocalTime)
+        {
+        }
+
+        /// <summary>Performs application-defined tasks associated with freeing, releasing, or resetting unmanaged resources.</summary>
+        /// <filterpriority>2</filterpriority>
+        public void Dispose()
+        {
+            DataConsolidated = null;
+            _dataConsolidatedHandler = null;
+        }
+
+        /// <summary>
+        /// Event invocator for the DataConsolidated event. This should be invoked
+        /// by derived classes when they have consolidated a new piece of data.
+        /// </summary>
+        /// <param name="consolidated">The newly consolidated data</param>
+        protected void OnDataConsolidated(VolumeRenkoBar consolidated)
+        {
+            DataConsolidated?.Invoke(this, consolidated);
+            _currentBar = consolidated;
+            _dataConsolidatedHandler?.Invoke(this, consolidated);
+            Consolidated = consolidated;
+        }
+
+        private decimal Next(IBaseData data, decimal volume)
+        {
+            while (volume >= BarSize)
+            {
+                var wicko = new VolumeRenkoBar(data.Symbol, OpenOn, CloseOn, BarSize, OpenRate, HighRate, LowRate, CloseRate, BarSize);
+                volume -= BarSize;
+
+                OnDataConsolidated(wicko);
+
+                OpenOn = CloseOn;
+                OpenRate = CloseRate;
+                HighRate = CloseRate;
+                LowRate = CloseRate;
+            }
+
+            return volume;
+        }
+    }
+
+    /// <summary>
+    /// Provides a type safe wrapper on the RenkoConsolidator class. This just allows us to define our selector functions with the real type they'll be receiving
```

Comment:
Given `VolumeRenkoConsolidator` is currently not generic it wouldn't make sense to do this?